pulumi/python/infrastructure/kubeconfig/__main__.py

```diff
@@ -1,40 +1,56 @@
 import pulumi
 import os
+import base64
 from kic_util import pulumi_config
 
-
 config = pulumi.Config('kubernetes')
 
+
 # For AWS
 def aws_project_name_from_project_dir(dirname: str):
     script_dir = os.path.dirname(os.path.abspath(__file__))
     project_path = os.path.join(script_dir, '..', '..', '..', 'python', 'infrastructure', 'aws', dirname)
     return pulumi_config.get_pulumi_project_name(project_path)
 
+
 # For Digital Ocean
 def do_project_name_from_project_dir(dirname: str):
     script_dir = os.path.dirname(os.path.abspath(__file__))
     project_path = os.path.join(script_dir, '..', '..', '..', 'python', 'infrastructure', 'digitalocean', dirname)
     return pulumi_config.get_pulumi_project_name(project_path)
 
-#
-# There are two paths currently available; if the user has requested that we stand up MARA on AWS
-# we pursue one route. If they request that a kubeconfig file be used, the second route is chosen.
-#
-# The difference is in where the information about the cluster is pulled from; if AWS is chosen the
-# data is pulled from the ../aws/eks directory. If Kubeconfig is chosen the information is pulled from
-# the configuration file under /config/pulumi.
-#
-# In both cases, this project is used to reference the kubernetes cluster. That is, this project exports
-# the variables used for cluster connection regardless of where it pulls them from (AWS project or kubeconfig).
-#
+
+# For Linode
+def lke_project_name_from_project_dir(dirname: str):
```

Comment:
If we have three implementations here defined in three functions, consider using a class to define per-provider implementations.

Reply:
Done